crates/typst/src/geom/gradient.rs

```diff
@@ -0,0 +1,1089 @@
+///
+/// ```example
+/// #rect(width: 100pt, height: 20pt, fill: gradient.linear(..gradient.mako))
+/// ```
+///
+/// ### Vlag
+///
+/// The [`vlag`]($gradient.vlag) gradient is a light blue to white to red
+/// gradient that is perceptually uniform. Vlag does not take any parameters.
+///
+/// ✅ This color map is suitable for data visualization.
+///
+/// ```example
+/// #rect(width: 100pt, height: 20pt, fill: gradient.linear(..gradient.vlag))
+/// ```
+///
+/// ### Icefire
+///
+/// The [`icefire`]($gradient.icefire) gradient is a light teal to black to yellow
+/// gradient that is perceptually uniform. Icefire does not take any parameters.
+///
+/// ✅ This color map is suitable for data visualization.
+///
+/// ```example
+/// #rect(width: 100pt, height: 20pt, fill: gradient.linear(..gradient.icefire))
+/// ```
+///
+/// ### Flare
+///
+/// The [`flare`]($gradient.flare) gradient is an orange to purple gradient
+/// that is perceptually uniform. Flare does not take any parameters.
+///
+/// ✅ This color map is suitable for data visualization.
+///
+/// ```example
+/// #rect(width: 100pt, height: 20pt, fill: gradient.linear(..gradient.flare))
+/// ```
+///
+/// ### Crest
+///
+/// The [`crest`]($gradient.crest) gradient is a blue to white to red gradient
+/// that is perceptually uniform. Crest does not take any parameters.
+///
+/// ✅ This color map is suitable for data visualization.
+///
+/// ```example
+/// #rect(width: 100pt, height: 20pt, fill: gradient.linear(..gradient.crest))
+/// ```
+///
+/// ### Regarding presets like "jet" and "parula"
+///
+/// - [Jet](https://jakevdp.github.io/blog/2014/10/16/how-bad-is-your-colormap/)
+///   is not a good color map, as it is not perceptually uniform. As such,
+///   it is not color blind friendly and should not be used for data visualization,
+///   due to which it is not included in Typst.
+/// - [Parula](https://www.mathworks.com/help/matlab/ref/parula.html)
+///   is a good color map included in MATLAB, but it is not included as
+///   a preset in Typst. This is because it is owned by MathWorks and is not public.
+///
+#[ty(scope)]
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub enum Gradient {
+    Linear(LinearGradient),
+}
+
+impl Debug for Gradient {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Linear(linear) => linear.fmt(f),
+        }
+    }
+}
+
+#[scope]
+impl Gradient {
+    pub const SPECTRAL: fn() -> Array = spectral;
+    pub const VIRIDIS: fn() -> Array = viridis;
+    pub const INFERNO: fn() -> Array = inferno;
+    pub const MAGMA: fn() -> Array = magma;
+    pub const PLASMA: fn() -> Array = plasma;
+    pub const ROCKET: fn() -> Array = rocket;
+    pub const MAKO: fn() -> Array = mako;
+    pub const VLAG: fn() -> Array = vlag;
+    pub const ICEFIRE: fn() -> Array = icefire;
+    pub const FLARE: fn() -> Array = flare;
+    pub const CREST: fn() -> Array = crest;
```

Comment:
It feels wrong to me that the presets are _arrays,_ not gradients. I guess it has to be that way because of other configuration and gradient kind etc., but still wanted to mention it.

Reply:
I agree with you, I thought of making them sort of constructors of their own, but it would have lead to so much duplication. I am open to any ideas regarding that!

Reply:
Perhaps the gradient constructors could accept other gradients as a parameter as well, automatically inheriting their properties? That way, the presets would have some good defaults, and would be easily customizable with `gradient.linear(rainbow, something_else, ...)`. (Another possibility would be to only inherit the stops from the gradient, but I'm not sure if that could be considered inconsistent.)

Reply:
That simply does not work as there is no « default » gradient and once the other gradients kinds are added that won’t make much sense. One solution could be to add a `colormap` type and make them as such (on which you can call `.stops()` and which can be taken directly as an argument, but otherwise it’s not doable in a nicer way than an array.